pkg/collector/py/datadog_agent.go

```diff
@@ -71,6 +71,22 @@ func Headers(self *C.PyObject, args *C.PyObject) *C.PyObject {
 
 		C.PyDict_SetItem(dict, pyKey, pyVal)
 	}
+
+	// some checks need to add an extra header when they pass `http_host`
+	if kwargs != nil {
+		cKey := C.CString("http_host")
+		// in case of failure, the following doesn't set an exception
+		// pyHttpHost is borrowed
+		pyHTTPHost := C.PyDict_GetItemString(kwargs, cKey)
```

Comment:
The cpython docs recommends using `PyArg_ParseTupleAndKeywords` to parse `kwargs` (https://docs.python.org/2/c-api/structures.html#METH_KEYWORDS), but I had a quick look at the implementation and all it would do for us in our case here would be to check that `kwargs` is indeed a dict, and unless something is horribly wrong with the interpreter it's a safe assumption to make 👍

Reply:
Cool, we already assume args are safe for functions accepting `METH_VARARGS`, I'm ok if we want to add calls to `PyArg_ParseTupleAndKeywords` and `PyArg_ParseTuple` where appropriate, in case I'd go with a dedicated PR.